rdt/transformers/categorical.py

```diff
@@ -192,7 +195,12 @@ def _reverse_transform(self, data):
                 labels.append(key)
 
         result = pd.cut(data, bins=bins, labels=labels, include_lowest=True)
-        return result.replace(nan_name, np.nan).astype(self.dtype)
+        result = result.replace(nan_name, np.nan)
+
+        if convert_to_float:
+            return result.astype(float)
+
+        return result.astype(self.dtype)
```

Comment:
I think instead of storing if the type needs to be converted to float we can just wrap the conversion in a try. If it fails, then we can see if the dtype is int and convert to float. Otherwise we just error. Then the `check_nan_in_transform` function can just be used to warn

Reply:
Thanks for the advice, I made the change